### ISSUE TITLE
FloatingFocusManager incorrect focus restore when the the reference unmounts before the floating element is closed

### ISSUE THREAD

**Issue by roigiladi** [floating-ui/floating-ui#3029]
**Describe the bug**
when `FloatingFocusManager` unmounts and it has `returnFocus={true}` - if the reference element that opened it is no longer available in the DOM (because it is conditionally rendered) then focus moves to a random (the first?) focusable element.

in my case - I have a button that opens a menu and when one of the items inside of it is clicked - I render a different component which doesn't contain the button (i.e: the reference element) so it's no longer available in the DOM.
when this happens, the focus moves to another element in my page which then runs some problematic side effects (due to an `onFocus` event handler on the now focused element) which results in a buggy UX

This behavior felt incorrect to me, but if it's actually the intended behavior then I can simply pass `returnFocus={false}` wherever i need

**To Reproduce**
https://codesandbox.io/p/sandbox/amazing-vaughan-ywsm5s?workspaceId=6aebd151-ce60-4ebd-b309-b7331d6bb1ed

1. Instructions in codesandbox

**Expected behavior**
In case the reference element is not available anymore when returning the focus, then do nothing (i think browsers default to `document.body`)

**Context:**

- OS: macOS
- Browser: Chrome
- Version: 0.26.23

Thanks!

**Comment by atomiks** [comment]
This was changed in https://github.com/floating-ui/floating-ui/pull/2946. If the ref/trigger gets removed, and focus returns to the body, then it gets "lost" in terms of tab index context if the floating element is portaled. By returning it to the nearest tabbable element, the document tab index is preserved regardless of portaling. 

One potential alternative is to immediately call `.blur()` once the nearest tabbable has been focused (I suppose you can actually do this externally, as a workaround). This makes portaled floating elements preserve the tab index like non-portaled ones and places focus on the body

**Comment by atomiks** [comment]
I believe I found a much better way to handle this issue that doesn't return focus to unrelated elements which may fire their event handlers undesirably. 

We can use a fallback hidden placeholder node, rendered inside `FloatingPortal` (this is the only case where it's necessary to begin with) to focus then blur immediately. Focus tab index context is preserved but the body gets the focus. The next/previous tab will focus the relevant elements that were near the reference element before it got removed.
